configure.py

```diff
@@ -60,55 +71,53 @@ def main(args):
     #llvm_projects.append("iree_dialects")
     llvm_configure_args.append(
         f"-DLLVM_EXTERNAL_IREE_DIALECTS_SOURCE_DIR={iree_dialects_path}")
-
-  # Detect LLVM.
-  if args.llvm_path:
-    llvm_path = os.path.abspath(args.llvm_path)
-    print(f"-- Using explicit llvm-project path: {llvm_path}")
-  elif llvm_path:
-    print(f"-- Using inferred llvm-project path: {llvm_path}")
   else:
-    llvm_path = os.path.join(args.repo_root, "..", "llvm-project")
+    # Detect LLVM.
+    # As per "Building without IREE" instructions:
+    # You **must** checkout [llvm-project](https://github.com/llvm/llvm-project) at a
+    # compatible commit next to this directory.
+    llvm_path = os.path.join(repo_root, "..", "llvm-project")
```

Comment:
This is problematic: I have different clones of llvm, and llvm-project is not the one I am using for this. Can we use an env var also for the path to LLVM? Or at least have the path to be specified as an option?

Reply:
I'll have to defer to @stellaraccident on this as the README and the script were inconsistent.
This has to do with how external projects build with LLVM and I think there were changes recently.
An easy way would be for you to put a symlink and call it a day.

Reply:
It's your project to decide. I personally prefer having options to specify the paths like the script originally had. I don't have a strong reason -- just a lot of time making scripts like this more general for weird situations. If it were me, I would use the side by side checkout thing as a default but slow override with options.

Reply:
I don't have a strong opinion on this, I would just like consistency between LLVM, the README, what configure.py does and my ability to build locally. 

The current proposed changes are my iteration on it as I was not able to reconcile things otherwise.
I am happy to revert to a prior version if all things agree.

It is my understanding we have the constraint that iree_llvm_sandbox, llvm_project and iree must be silblings under a common parent root (or we had that constraint until recently). This seems to directly contradict being able to use iree/llvm-project, so this can't be true.

Anyway, I am looking for advice / what to revert while still being able to build and run the project.

Reply:
The main problem is that in the same folder I have at least three llvm clones, and one of them is named llvm-project, but it is quite old, so I was getting weird errors (albeit the `configure.py` was still detecting the `llvm-project` folder) . I agree with @stellaraccident , having an additional option to override the default llvm-project path should be simple enough and unharmful. I don't want to block this PR though. I can add it later in a different PR

Reply:
My guidance would be: the side by side constraint is a useful default, but explicit paths/options always gets used eventually for advanced cases.

The property that you really want is: check things out like this, run script without options, success. Having knobs to control it for advanced cases usually is required eventually, and there is nothing in the build setup that *requires* these projects live side by side.

Reply:
`there is nothing in the build setup that requires these projects live side by side` ok I think this is the information I was hung up on, I should then be able to update more of the README and less of the configure script. I'll start a new PR.